Cross-compiling for musl still fails:
```
[yuka@nixos:~/proj/nixfiles]$ nix build 'git+https://github.com/Mindavi/nixpkgs?ref=tpm2-tss/cross#pkgsStatic.tpm2-tss'
error: builder for '/nix/store/rmj6cy9sfrj0v57rnd1jzj5g9sl4ybfb-cmocka-static-x86_64-unknown-linux-musl-1.1.5.drv' failed with exit code 2;
       last 10 log lines:
       > [ 18%] Building C object example/mock/uptime/CMakeFiles/uptime.dir/uptime.c.o
       > [ 22%] Linking C executable uptime
       > [ 22%] Built target uptime
       > [ 27%] Linking C shared library libcmocka.so
       > /nix/store/2hnyzi21y3gp9l1m48sgxzz1s2n0d61p-x86_64-unknown-linux-musl-binutils-2.35.1/bin/x86_64-unknown-linux-musl-ld: /nix/store/syy43yipps3sx8m9lva3l7fxf9lqpdh2-x86_64-unknown-linux-musl-stage-final-gcc-10.3.0/lib/gcc/x86_64-unknown-linux-musl/10.3.0/crtbeginT.o: relocation R_X86_64_32 against hidden symbol `__TMC_END__' can not be used when making a shared object
       > /nix/store/2hnyzi21y3gp9l1m48sgxzz1s2n0d61p-x86_64-unknown-linux-musl-binutils-2.35.1/bin/x86_64-unknown-linux-musl-ld: /nix/store/syy43yipps3sx8m9lva3l7fxf9lqpdh2-x86_64-unknown-linux-musl-stage-final-gcc-10.3.0/lib/gcc/x86_64-unknown-linux-musl/10.3.0/crtend.o: relocation R_X86_64_32 against `.ctors' can not be used when making a shared object; recompile with -fPIC
       > collect2: error: ld returned 1 exit status
       > make[2]: *** [src/CMakeFiles/cmocka.dir/build.make:97: src/libcmocka.so.0.7.0] Error 1
       > make[1]: *** [CMakeFiles/Makefile2:239: src/CMakeFiles/cmocka.dir/all] Error 2
       > make: *** [Makefile:166: all] Error 2
       For full logs, run 'nix log /nix/store/rmj6cy9sfrj0v57rnd1jzj5g9sl4ybfb-cmocka-static-x86_64-unknown-linux-musl-1.1.5.drv'.
error (ignored): error: cannot unlink '/tmp/nix-build-ncurses-6.2-x86_64-unknown-linux-musl.drv-1': Directory not empty
error: 1 dependencies of derivation '/nix/store/mghirs4l99wdyvlwvgih8gissq1slsrx-tpm2-tss-static-x86_64-unknown-linux-musl-3.0.3.drv' failed to build

[yuka@nixos:~/proj/nixfiles]$

--- a/pkgs/development/libraries/tpm2-tss/default.nix
+++ b/pkgs/development/libraries/tpm2-tss/default.nix
@@ -26,11 +26,21 @@ stdenv.mkDerivation rec {
   nativeBuildInputs = [
     autoreconfHook autoconf-archive pkg-config doxygen perl
   ];
-  buildInputs = [ openssl json_c curl libgcrypt ];
-  checkInputs = [
-    cmocka uthash ibm-sw-tpm2 iproute2 procps_pkg which
+
+  # cmocka is checked / used(?) in the configure script
+  # when unit and/or integration testing is enabled
+  buildInputs = [ openssl json_c curl libgcrypt uthash ]
+    # cmocka doesn't build with pkgsStatic, and we don't need it anyway
+    # when tests are not run
+    ++ lib.optionals (stdenv.buildPlatform == stdenv.hostPlatform) [
+    cmocka
   ];
 
+  checkInputs = [
+    cmocka which openssl procps_pkg iproute2 ibm-sw-tpm2
+  ];
+
+  strictDeps = true;
   preAutoreconf = "./bootstrap";
 
   enableParallelBuilding = true;
@@ -49,7 +59,7 @@ stdenv.mkDerivation rec {
       --replace '@PREFIX@' $out/lib
   '';
 
-  configureFlags = [
+  configureFlags = lib.optionals (stdenv.buildPlatform == stdenv.hostPlatform) [
     "--enable-unit"
     "--enable-integration"
   ];

